Sources/CvSelectionGroupAI.cpp

```diff
@@ -757,10 +709,13 @@ CvUnit* CvSelectionGroupAI::AI_getBestGroupAttacker(const CvPlot* pPlot, bool bP
 		// Recalculate, this time modifying the predicted hitpoints that will result
 		// if the battle occurs.  This will return a better odds estimate than we
 		//	had previously, but should not change the unit choice
-		iBestOdds = pBestUnit->AI_attackOdds(pPlot, bPotentialEnemy, ppDefender);
+		iUnitOdds = pBestUnit->AI_attackOdds(pPlot, bPotentialEnemy, ppDefender);
 	}
-	
-	iUnitOdds = iBestOdds;
+	else
+	{
+		iUnitOdds = iBestValue;
```

Comment:
Odds aren't the same as the value before the changes, are you sure this is correct?

Reply:
This bit is rather odd actually.  iUnitOdds doesn't seem to be used. If pBestUnit->AI_attackOdds(pPlot, bPotentialEnemy, ppDefender) does nothing exceptreturn something, the whole part can be removed.